base-template.sh

```diff
@@ -759,15 +759,18 @@ show_prompt() {
     fi
 
     # Try to read from `dev/tty` if available.
-    # otherwise just read from stdin.
+    # Our stdin is never a tty (either a pipe or /dev/null when called
+    # from git), so read from /dev/tty, our controlling terminal.
+    # However, only do this when stdout *is* a tty, otherwise it is
+    # likely we have no controlling terminal and reading from /dev/tty
+    # would fail with an error.
     printf "%s %s [%s]:" "$TEXT" "$HINT_TEXT" "$SHORT_OPTIONS"
-    if [ -t 0 ]; then
-        # shellcheck disable=SC2229
-        read -r "$VARIABLE" </dev/tty
-    else
-        # shellcheck disable=SC2229
-        read -r "$VARIABLE"
-    fi
+    # shellcheck disable=SC2229
+    read -r "$VARIABLE" </dev/tty 2>/dev/null
```

Comment:
Why do we need to suppress the error here? Could we just let it through? It might help working out problems better.

Reply:
@gabyx can you have a look at this please? This PR should be good to go otherwise.

Reply:
If you don't suppress this error, I think you're essentially just reverting #81 and you are back to the problem described there (which I *think* is only that an error message is shown, I do not think that #81 actually fixed anything otherwise).

Reply:
Ah yes, I see. Not super awesome to suppress the error but I guess let's try and see. @gabyx happy for this PR to go in? We're not waiting on anything else for this, right?

Reply:
(Hm, started typing this comment before the merge and then got called away. In case it's still usedful, I'll finish it anyway)

The alternative that also seems reasonable to me (which I suggested before), is to check whether stdout is a tty (`[ -t 1 ]`) and if not, just not read at all (maybe not even show the prompt) and return an error. This probably ties in with #84 a bit as well.

Or, I realize now, you could also maybe first try opening /dev/tty separately and supress output separately there. E.g. `if true < /dev/tty 2> /dev/null; then` might work to check if opening `/dev/tty` would work without suppressing (much) other errors. Ideally you would just do a libc `open` call and check the return code (for a specific error code, `ENODEV` I  think), but I don't think sh/bash allows that.

Reply:
> Or, I realize now, you could also maybe first try opening /dev/tty separately and supress output separately there. E.g. if true < /dev/tty 2> /dev/null; then might work to check if opening /dev/tty would work without suppressing (much) other errors.

Hm, that sounds good, it might be worth to try. @gabyx any chance you could test it in the environment this was originally broken for you?